### PR TITLE
Prefix per server

### PR DESCRIPTION
**Pull request by itsTheFae**
- [x] I have tested my changes against the `dev` branch (the latest developmental version), and this pull request is targeting that branch as a base
- [x] I have tested my changes on Python 3.10.12
- [x] I have ensured my code is formatted using [Black](https://github.com/psf/black)

----

### Description

This PR adds support for setting a command prefix per guild/server.  It adds one new option `EnableOptionsPerGuild` and a new command `setprefix` to manage the feature.  
This feature allows just about anything without spaces to be set as a command prefix, including Emoji.  Only one verb cannot be set, as `setprefix clear` is used to clear any currently set prefix.
The default prefix in `options.ini` is used when no server-specific prefix is set.  If per-guild options are not enabled the `setprefix` command will raise an error.  

Additionally, this feature will remember a limited number of prefix changes so the `clean` command can take care of messages prior to a prefix change.   This history of prefixes does not persist through restarts.

### Related issues (if applicable)

#2288

### CODE REVIEW

File: config/example_options.ini [hunk 1]
```diff
@@ -166,6 +166,9 @@ SearchList = no
 # Set the amount of search results returned by default.
 DefaultSearchResults = 3
 
+# Allow per-guild command prefix override via 'setprefix' command.
+EnableOptionsPerGuild = no
```
Comment (DestinyofYeet) [inline config/example_options.ini]: Personally I'd think something like `EnablePrefixPerGuild` would be better.

**Review by DestinyofYeet — COMMENTED** [review]
Looks good

File: config/example_options.ini (hunk 1 above)
Comment (itsTheFae) [reply config/example_options.ini]: I agree.  I started this intending to add more options but ended up just doing command prefix to keep it focused.

**Comment by itsTheFae** [comment]
This should be fully ready to go now. :+1:

**Pull request merged**